ccmbioinfo/lfs
# CARDS-1336: Serialize survey responses to EPIC-friendly json format

POC for the [CARDS-1336](https://phenotips.atlassian.net/browse/CARDS-1336)

To test with 
- start with ` ./start_cards.sh -P prems -f mvn:io.uhndata.cards/cards-epic-integration/0.9-SNAPSHOT/slingosgifeature`
- in the Questionnaire editor add `ExternalLink` to the questionnaire and some questions (**label** has to be `epic`)
- fill in the Questionnaire form and the create the corresponsing `Visit` form for it
- check the Questionnaire form link with URL suffix `.toEpic.deep.json`. 
For example: `http://localhost:8080/Forms/84c7c113-4d7e-4b88-b890-4b20240abacf.toEpic.deep.json`

In the generated JSON you should see:
```
{
  "resourceType": "QuestionnaireResponse",
  "questionnaire": "Questionnaire/<your created ExternalLink>",
  "status": "in-progress", // if Visit form is not submitted
  "subject": {
       "identifier": {
                "system": "<placeholder: CarePlan.activity.reference.type>",
                "value": "<placeholder: CarePlan.activity.reference.identifier.value -> CSN ID for an Appointment>"
            }
  },
  // for answers:
  [
      {
          "linkId": "<placeholder: ID of the Question ID Type>|<your created ExternalLink>",
          "text": "<question text>",
                  "answer": [
                      {
                           "valueString": "<question answer>"
                       }
                    ]
      }, 
  ]
}
```


[CARDS-1336]: https://phenotips.atlassian.net/browse/CARDS-1336?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
